### ISSUE TITLE
Kubernetes ServiceAccounts Integration with NATS Authorization

### ISSUE THREAD

**Issue by wallyqs** [nats-io/nats-operator#52]
In last release, it was added support for authorization using a custom secret where the credentials are defined in JSON, but we might be able to simplify this by using the service accounts already present in Kubernetes.

Below a full example of a manifest creating 2 pods, one binding to the default service account and the other to a new service account, then a couple of custom `ServiceRoles` (new CRD managed by the operator) created to define the pub/sub permissions. A NATS client running in one of the pods would be able to use the token provided by the service account, and in case of changes the configuration would be reloaded with the new authorization rules.

```sh
# Container using a new service account
nats-sub -s nats://demo-nats-service-account:`cat /var/run/secrets/kubernetes.io/serviceaccount/token`@demo-nats:4222 SANDBOX.hello

# Container using the default service account
nats-sub -s nats://default:`cat /var/run/secrets/kubernetes.io/serviceaccount/token`@demo-nats:4222 SANDBOX.hello
```

Example manifest of how this would work:
```yaml
---
apiVersion: nats.io/v1alpha3
kind: NatsCluster
metadata:
  name: demo-nats
spec:
  size: 3
  version: "1.1.0"
  pod:
    enableConfigReload: true
  auth:
    enableServiceAccounts: true
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: demo-nats-service-account
---
apiVersion: nats.io/v1alpha3
kind: ServiceRole
metadata:
  name: demo-nats-role
  namespace: default

  # Specifies which NATS cluster will be mapping this account,
  # (have to create a service role with permission per cluster).
  labels:
    nats_cluster: demo-nats
spec:
  serviceAccountName: demo-nats-service-account
  permissions:
    publish: ["foo.*", "foo.bar.quux"]
    subscribe: ["foo.bar"]
---
apiVersion: nats.io/v1alpha3
kind: ServiceRole
metadata:
  name: demo-nats-default-role
  namespace: default

  # Specifies which NATS cluster will be mapping this account,
  # (have to create a service role with permission per cluster).
  labels:
    nats_cluster: demo-nats
spec:
  serviceAccountName: default
  permissions:
    publish: ["SANDBOX.>"]
    subscribe: ["SANDBOX.>"]
---
apiVersion: nats.io/v1alpha3
kind: ServiceRole
metadata:
  name: default
  namespace: default
  labels:
    nats_cluster: demo-nats
spec:
  serviceAccountName: demo-nats-service-account
  permissions:
    publish: ["foo.*"]
    subscribe: ["foo.bar"]
---
apiVersion: v1
kind: Pod
metadata:
  name: demo-nats-client-pod
spec:
  serviceAccountName: demo-nats-service-account
  restartPolicy: Never
  containers:
    - name: nats-ops
      command: ["/bin/sh"]
      image: "wallyqs/nats-ops:latest"
      tty: true
      stdin: true
      stdinOnce: true
---
apiVersion: v1
kind: Pod
metadata:
  name: demo-nats-client-pod-default-account
spec:
  # No account means using the default service token
  restartPolicy: Never
  containers:
    - name: nats-ops
      command: ["/bin/sh"]
      image: "wallyqs/nats-ops:latest"
      tty: true
      stdin: true
      stdinOnce: true
```

**Comment by wallyqs** [comment]
Rather than using the service account secret, instead looking into using the TokenRequest API that was built into Kubernetes v1.10 to issue a token with a different audience so that can't be used against the API server.
https://github.com/nats-io/nats-operator/pull/58

**Comment by wallyqs** [comment]
Fixed via #58
